InternalDocs/interpreter.md

```diff
@@ -0,0 +1,376 @@
+
+The bytecode interpreter
+========================
+
+Overview
+--------
+
+This document describes the workings and implementation of the bytecode
+interpreter, the part of python that executes compiled Python code. Its
+entry point is in
+[Python/ceval.c](https://github.com/python/cpython/blob/main/Python/ceval.c).
+
+At a high level, the interpreter consists of a loop that iterates over the
+bytecode instructions, executing each of them via a switch statement that
+has a case implementing each opcode. This switch statement is generated
+from the instruction definitions in
+[Python/bytecodes.c](https://github.com/python/cpython/blob/main/Python/bytecodes.c)
+which are written in
+[a DSL](https://github.com/python/cpython/blob/main/Tools/cases_generator/interpreter_definition.md)
+developed for this purpose.
+
+Recall that the [Python Compiler](https://github.com/python/cpython/blob/main/InternalDocs/compiler.md)
+produces a [`CodeObject`](https://github.com/python/cpython/blob/main/InternalDocs/code_object.md),
```

Comment:
I think you might be able to get away with just this (also applies to a bunch of other links in this PR):

```suggestion
Recall that the [Python Compiler](../compiler.md) produces a [`CodeObject`](../code_object.md),
```

Reply:
I guess I could also use relative links to the code (like `../Python/bytecodes.c`).

Reply:
Try it and see how it renders? You can get a good preview of how GitHub will render it by pushing to this PR branch, then clicking this "View file" button here from the "files changed" tab:

![image](https://github.com/user-attachments/assets/ea3f0b5c-9844-4f92-b243-3b7192d63e12)